server/src/core/user/user-service.ts

```diff
@@ -150,27 +160,86 @@ export class Users implements UserService {
     }
   }
 
-  async nominateBeneficiary(args: UserNominateArgs): Promise<User> {
-    const { phone, email, nominator, name } = args;
+  async nominate(args: UserNominateArgs): Promise<Invitation> {
     validators.validatesNominate(args);
+    const { phone, email, nominatorId, nominatorName, name, role } = args;
+    
+    try {
+      const nominatorUser = await this.collection.findOne({ _id: nominatorId });
+      if (nominatorUser && nominatorUser.roles.includes('beneficiary')) 
+        throw createBeneficiaryActivationFailedError(messages.ERROR_USER_CANNOT_ADD_BENEFICIARY);
+      else if (!nominatorUser) throw createResourceNotFoundError(messages.ERROR_USER_NOT_FOUND);
+
+      const inviteeUser = await this.collection.findOne({ phone });
+      const invitationArgs: InvitationCreateArgs = {
+        invitorId: nominatorId,
+        invitorName: nominatorName,
+        inviteeName: name,
+        inviteePhone: phone,
+        inviteeEmail: email,
+        inviteeRole: role,
+        hasAccount: inviteeUser ? true : false
+      }
+      const invitation = await this.invitations.create(invitationArgs);
+      this.eventBus.emitUserInvitationCreated({
+        recipientName: invitation.inviteeName,
+        recipientPhone: invitation.inviteePhone,
+        senderName: invitation.invitorName,
+        role: invitation.inviteeRole,
+        invitationId: invitation._id
+      });
+      return invitation;
+    }
+    catch (e) {
+      if (e instanceof AppError) throw e;
+      throw createDbOpFailedError(e.message);
+    }
+  }
 
+  async activate(args: UserActivateArgs): Promise<User> {
+    validators.validatesActivate(args);
+    const { invitationId } = args;
     try {
-      const nominatorUser = await this.collection.findOne({ _id: nominator });
-      if (!nominatorUser) throw createResourceNotFoundError(messages.ERROR_USER_NOT_FOUND);
+      const invitation = await this.invitations.get(invitationId);
+      if (invitation.status !== 'accepted') 
+        throw createResourceNotFoundError(messages.ERROR_INVITATION_NOT_FOUND);
+      
+      const args = {
+        phone: invitation.inviteePhone,
+        name: invitation.inviteeName,
+        email: invitation.inviteeEmail,
+        nominatorId: invitation.invitorId,
+      }
+      if (invitation.inviteeRole === 'beneficiary') return this.activateBeneficiary(args);
+      else if(invitation.inviteeRole === 'middleman') return this.activateMiddleman(args);
```

Comment:
Please create if-else statement with proper curly braces for better readability.
Also, since there are return statements, there's no need for the `else`, you could just have two if statements.